assets/model_monitoring/components/src/shared_utilities/io_utils.py

```diff
@@ -126,11 +126,26 @@ def _verify_mltable_paths(mltable_path: str, ws=None, mltable_dict: dict = None)
             raise InvalidInputError(f"Invalid or unsupported path {path_val} in MLTable {mltable_path}") from iie
 
 
-def _write_mltable_yaml(mltable_obj, dest_path):
+def _write_mltable_yaml(mltable_obj, dest_path, file_system=None):
     try:
-        content = yaml.dump(mltable_obj, default_flow_style=False)
-        store_url = StoreUrl(dest_path)
-        store_url.write_file(content, 'MLTable', True)
+        import os
+        import uuid
+        from azureml.fsspec import AzureMachineLearningFileSystem
+
+        folder_name = str(uuid.uuid4())
```

Comment:
nit: python have a tempfile package to handle temp file, better than creating it manually with uuid. can refer model_data_collector_preprocessor.py for the usage.
it is nit, we can do it later if you want this PR to merge quickly

Reply:
need to fix next time, actually Andrei suggests uuid :)